src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -465,11 +465,21 @@ class HealthCertificateService {
 
 			switch result {
 			case .success(let healthCertificateBase45):
-				Log.info("[HealthCertificateService] Certificate assembly succeeded", log: .api)
 
 				remove(testCertificateRequest: testCertificateRequest)
-				registerHealthCertificate(base45: healthCertificateBase45)
-				completion?(.success(()))
+				let registerResult = registerHealthCertificate(base45: healthCertificateBase45)
+
+				switch registerResult {
+				case .success:
+					Log.info("[HealthCertificateService] Certificate assembly succeeded", log: .api)
+					completion?(.success(()))
```

Comment:
I think the `remove(testCertificateRequest: testCertificateRequest)` needs to move down here so we don't delete the request in the error case. I think there was a problem in the unit tests that made me move it above the registration, we can discuss it if there are problems with this change.

Reply:
moved